alerter.cpp

```diff
@@ -1,32 +1,62 @@
 #include <iostream>
 #include <assert.h>
+#include <string>
 
 int alertFailureCount = 0;
+#define TEMPTHRESHOLD 200.0
+#define SUCCESS 200
+#define FAILURE 500
 
-int networkAlertStub(float celcius) {
+int testNetworkAlertStub(float celcius)
+{
     std::cout << "ALERT: Temperature is " << celcius << " celcius.\n";
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if (celcius > TEMPTHRESHOLD)
+    {
+        return FAILURE;
+    }
+    else
+    {
+        return SUCCESS;
+    }
+}
+
+int testProductionAlertStub(float celcius)
+{
+    std::cout << "ALERT: Temperature is " << celcius << " celcius.\n";
+    // Threshold needs to be defined for production environment, currently returning SUCCESS in all cases.
+    return SUCCESS;
+}
+
+float convertFarenehitToCelcius(float f)
+{
+    return((f - 32) * 5 / 9);
 }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
+void alertInCelcius(const char* env, float farenheit)
+{
+    int returnCode;
+    std::string environment(env);
+    float celcius = convertFarenehitToCelcius(farenheit);
+    if (environment == "Test environment")
+        returnCode = testNetworkAlertStub(celcius);
+    else if (environment == "Production environment")
+        returnCode = testProductionAlertStub(celcius);

```

Comment:
Stub should not be used in the production code naming. Stubs mean some testing infrastructure.